neqo-crypto/src/agentio.rs

```diff
@@ -222,13 +222,13 @@ pub struct AgentIo {
 }
 
 impl AgentIo {
-    pub const fn new() -> Self {
+    pub fn new() -> Self {
         Self {
             input: AgentIoInput {
                 input: null(),
                 available: 0,
             },
-            output: Vec::new(),
+            output: Vec::with_capacity(1500), // Pre-allocate for typical TLS record output
```

Comment:
Well, this is also wrong.  It's the analogue of the pre-allocation we'd use for CRYPTO frames.  It is not record size.

Reply:
In the one test (`neqo-crypto::agent basic`) where this isn't zero, it's 238.